Assets/Code/src/Runtime/Match/MatchSimulation.cs

```diff
@@ -39,6 +56,16 @@ public class MatchSimulation : IMatchSimulation {
     }
   }
 
+  IMatchSimulation[] CreateSimulations(IMatchSimulation[] matchSimulations, params Type[] restrictions) {
+    var Simulation = new List<IMatchSimulation>();
+    foreach (var simComponent in matchSimulations) {
```

Comment:
This is reducible via Linq to `matchSimulations.Where(sim => restrictions.Contains(sim)).ToArray()`